internals/overlord/cmdstate/request.go

```diff
@@ -70,8 +70,14 @@ func Exec(st *state.State, args *ExecArgs) (*state.Task, ExecMetadata, error) {
 		return nil, ExecMetadata{}, errors.New("cannot use interactive mode without a terminal")
 	}
 
-	// Inherit the pebble daemon environment.
+	// Inherit the pebble daemon environment (except HOME and USER env vars)
 	environment := osutil.Environ()
+	for _, k := range []string{"HOME", "USER"} {
+		if _, ok := environment[k]; ok {
+			delete(environment, k)
```

Comment:
Per https://pkg.go.dev/builtin#delete, `delete(m, k)` is a no-op if `k` isn't present in `m`, so it's fine to just delete without the check. Also, since there are just two variables, it's probably simplest to just write them out. So we can change this whole block to:

```go
delete(environment, "HOME")
delete(environment, "USER")
```

Reply:
addressed in [88a02c5](https://github.com/canonical/pebble/pull/262/commits/88a02c5938164e08d2f8380297ce24727c695a49)